src/pkg/services/m365/api/contacts_pager.go

```diff
@@ -67,42 +66,14 @@ func (p *contactsFoldersPageCtrl) ValidModTimes() bool {
 	return true
 }
 
-// EnumerateContainers iterates through all of the users current
-// contacts folders, transforming each to a graph.CacheFolder, and calling
-// fn(cf).
-// Contact folders are represented in their current state, and do
-// not contain historical data.
+// EnumerateContainers retrieves all of the users current contact folders.
 func (c Contacts) EnumerateContainers(
 	ctx context.Context,
 	userID, baseContainerID string,
 	immutableIDs bool,
-	fn func(graph.CachedContainer) error,
-	errs *fault.Bus,
-) error {
-	var (
-		el  = errs.Local()
-		pgr = c.NewContactFoldersPager(userID, baseContainerID, immutableIDs)
-	)
-
-	containers, err := batchEnumerateItems(ctx, pgr)
-	if err != nil {
-		return graph.Stack(ctx, err)
-	}
-
-	for _, c := range containers {
-		if el.Failure() != nil {
-			break
-		}
-
-		gncf := graph.NewCacheFolder(c, nil, nil)
-
-		if err := fn(&gncf); err != nil {
-			errs.AddRecoverable(ctx, graph.Stack(ctx, err).Label(fault.LabelForceNoBackupCreation))
-			continue
-		}
-	}
-
-	return el.Failure()
+) ([]models.ContactFolderable, error) {
+	containers, err := batchEnumerateItems(ctx, c.NewContactFoldersPager(userID, baseContainerID, immutableIDs))
```

Comment:
calling this out mostly for discussion, but it seems like eventually we're going to want to switch to an implementation that doesn't require us to pass in a base folder ID. I played around a bit with the API and it seems that the beta version of the `contactFolders` endpoint returns all folders.

The catch there is that some folders have been deleted and thus have a different parent ID than the rest of the folders. The deleted folder is not something that we can get either it seems. It doesn't have a "well known" name nor can we get it with `contactFolders/<folder ID>`

Switching to the beta endpoint would get us nested folder backup support though since it's quite likely we're missing it right now

Reply:
added a note about it here: https://github.com/alcionai/corso/issues/3550